### PR TITLE
Test for functional argument passing

### PR DESCRIPTION
**Pull request by directionless**
Tests for https://github.com/tinygo-org/tinygo/pull/322

(And a .gitignore addition for llvm)

### CODE REVIEW

**Comment by directionless** [comment]
I suspect this should _fail_ tests, I think #322 merged into dev, not master.

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,5 @@ src/device/stm32/*.s
 src/device/sam/*.go
 src/device/sam/*.s
 vendor
+llvm
```
Comment (deadprogram) [inline .gitignore]: These exclusions should not be part of this commit/PR.

**Comment by deadprogram** [comment]
Hi @directionless thanks for the PR I changed the branch for it to `dev` as mentioned here: https://github.com/tinygo-org/tinygo/blob/master/CONTRIBUTING.md#how-to-use-our-github-repository

File: .gitignore (hunk 1 above)
Comment (directionless) [reply .gitignore]: Yeah, I was short cutting a bit. I'll pull them.

**Comment by aykevl** [comment]
Can you merge these tests in calls.go? By having fewer larger tests, the time to run them is reduced (certainly until #285 has been finished).
Also, can you use `println` instead of `fmt.Println`? By avoiding this fmt package, tests are much faster to execute because they compile faster. This test isn't about the fmt package but only about named functions.

I care a lot about fast tests because the slower the test, the less likely you are to run them.

**Comment by directionless** [comment]
Sure. I'm happy to merge this with `calls.go`.

My bias is also to faster tests, though in my other go projects I haven't found fewer files to be as much of a win. But I also use a lot of parallelization. I tossed a bunch of stuff into https://github.com/tinygo-org/tinygo/pull/337 around that. On my machine, it seems to cut the test time in half. I'm curious what you think of that approach.

**Comment by deadprogram** [comment]
Thanks for the additional tests @directionless now merging.

**Comment by aykevl** [comment]
Technically varargs are a matter for the frontend (https://godoc.org/golang.org/x/tools/go/ssa) because on the Go SSA level, all varargs are replaced with slices already so doesn't really need to care about this, but it's good to test it anyway to be sure.
